lib/mongo_compat_api.js

```diff
@@ -0,0 +1,325 @@
+          });
+        });
+      }
+    };
+
+    /**
+     * List all collections
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#listCollections
+     * @param filter Query to filter collections by
+     * @param options (optional) Optional settings
+     * @returns Cursor
+     */
+    this.listCollections = function (filter, options) {
+      var options = this.copyDefaultOptions(options || {});
+      return ditcher.doGetCollectionsListCursor(filter, options);
+    };
+
+    /**
+     * Return collections instance
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#collection
+     * @param name Query to filter collections by
+     * @param options (optional) Optional settings
+     * @param callback (optional) result callback
+     * @returns Collection instance if no callback passed
+     */
+    this.collection = function (name, options, callback) {
+      assert.ok(name, "collection name required");
+
+      var options = this.copyDefaultOptions(options || {});
+      options.type = name;
+
+      return ditcher.doGetCollectionInstance(options, callback);
+    };
+
+    /**
+     * Create an index on collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#createIndex
+     * @param name Name of the collection
+     * @param fieldOrSpec Field to use for the index
+     * @param options (optional) Options
+     * @param callback (optional) result callback
+     * @returns Promise if no callback passed
+     */
+    this.createIndex = function (name, fieldOrSpec, options, callback) {
+      assert.ok(name, "collection name required");
+      assert.ok(fieldOrSpec, "index spec required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      var options = this.copyDefaultOptions(options || {});
+      options.type = name;
+      options.index = fieldOrSpec;
+
+      if (callback) {
+        ditcher.doIndex(options, callback);
+      } else {
+        return new Promise(function (resolve, reject) {
+          ditcher.doIndex(options, function (err, result) {
+            if (err) {
+              return reject(err);
+            }
+
+            resolve(result);
+          });
+        });
+      }
+    };
+
+    /**
+     * Rename a collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#renameCollection
+     * @param fromCollection Current name
+     * @param toCollection New name
+     * @param options (optional) Options
+     * @param callback (optional) Result callback
+     */
+    this.renameCollection = function (fromCollection, toCollection, options, callback) {
+      assert.ok(fromCollection, "current collection name required");
+      assert.ok(toCollection, "new collection name required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      var options = this.copyDefaultOptions(options || {});
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "options" as this name is already used in declaration at line 167. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)